enumerable.rb

```diff
@@ -0,0 +1,125 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength,
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n < to_a.length
+      yield(self[n]) if is_a?(Array)
+      yield(keys[n], self[keys[n]]) if is_a?(Hash)
+      yield(to_a[n]) if is_a?(Range)
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index(&block)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    n = 0
+    while n < to_a.length
+      block.call(self[n], n) if is_a?(Array)
+      block.call(keys[n], self[keys[n]]) if is_a?(Hash)
+      block.call(to_a[n]) if is_a?(Range)
+      n += 1
+    end
+    self
+  end
```

Comment:
- #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self. Same here, you are returning the key but not the value, kindly return the value as well
- #my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self. The method is not returning the index when self is a range. Kindly make it happen